docs/source/pika-comparison.rst

```diff
@@ -0,0 +1,288 @@
+
+    channel.queue_declare(queue='task_queue', durable=True)
+
+    message = ' '.join(sys.argv[1:]) or "Hello World!"
+    channel.basic_publish(
+        exchange='',
+        routing_key='task_queue',
+        body=message,
+        properties=pika.BasicProperties(
+            delivery_mode=2,  # make message persistent
+        ))
+    print(" [x] Sent %r" % message)
+    connection.close()
+
+The code for running a worker::
+
+    import pika
+    import time
+
+    connection = pika.BlockingConnection(
+        pika.ConnectionParameters(host='localhost'))
+    channel = connection.channel()
+
+    channel.queue_declare(queue='task_queue', durable=True)
+    print(' [*] Waiting for messages. To exit press CTRL+C')
+
+
+    def callback(ch, method, properties, body):
+        print(" [x] Received %r" % body)
+        time.sleep(body.count(b'.'))
+        print(" [x] Done")
+        ch.basic_ack(delivery_tag=method.delivery_tag)
+
+
+    channel.basic_qos(prefetch_count=1)
+    channel.basic_consume(queue='task_queue', on_message_callback=callback)
+
+    channel.start_consuming()
+
+
+KiwiPy
+++++++
+
+And now, in kiwiPy.  The code for sending the same task:
+
+.. literalinclude:: examples/pika/new_task.py
+
+Here, compared to the Pika snippet, we see that we don't have to decide or know about an ``exchange``, ``routing_key``, ``properties``, ``channel`` or durability of the message.  Tasks queues in kiwiPy are always durable and the corresponding messages persistent.  The ``routing_key`` is simply the name of the queue that the user declared and an ``exchange`` is selected automatically for the user (set to ``kiwipy.tasks`` by default).
+
+Here we have explicitly created a task queue called ``task_queue``, however, even this has a default version if the user doesn't need multiple queues.  In which case the code would simply be::
+
+    with kiwipy.connect('amqp://localhost') as comm:
+        comm.task_send(message)
+        print(" [x] Sent %r" % message)
+
+Now to run a worker it's just:
+
+.. literalinclude:: examples/pika/worker.py
+
+This is fairly straightforward and the only decision the user has to make is about the prefetch count (the number of tasks and worker can be working on simultaneously).
+
+
+Topics
+------
+
+Moving on to the `topics`_ example (known as broadcasts with filters in kiwiPy).
+
+Pika
+++++
+
+The Pika code to emit a log topic is::
+
+    import pika
+    import sys
+
+    connection = pika.BlockingConnection(
+        pika.ConnectionParameters(host='localhost'))
+    channel = connection.channel()
+
+    channel.exchange_declare(exchange='topic_logs', exchange_type='topic')
+
+    routing_key = sys.argv[1] if len(sys.argv) > 2 else 'anonymous.info'
+    message = ' '.join(sys.argv[2:]) or 'Hello World!'
+    channel.basic_publish(
+        exchange='topic_logs', routing_key=routing_key, body=message)
+    print(" [x] Sent %r:%r" % (routing_key, message))
+    connection.close()
```

Comment:
would be good to preface these with `.. code:: python` to enable syntax highlighting. But actually would be even better to use `.. literalinclude` like you do for the `kiwipy` snippets